codegen/facelift/templates/Struct.template.cpp

```diff
@@ -95,6 +96,12 @@ void {{struct.name}}::deserialize(const QByteArray &array)
 
 {% endif %}
 
+{% if struct.treatAsByteArray %}
+bool {{struct.name}}::treatAsByteArray()
```

Comment:
The fact that a structure is serialized as a byte array or not is not a property of the structure itself. It belongs to the IPC layer so it has nothing to do in the definition of the structure itself.

Reply:
The "treatAsByteArray" is a structure annotation, adding that to the structure properties is only natural.
Moving it out needs introduction of a whole new layer for DBusIPC to be able to treat structs generically with no benefit what-so-ever.
Sorry I won't do that, those kind over-engineered decisions (like having all structs internal properties exposed as tuple for some other layer to be able to serialize it!!!) has made facelift a highly unnecessary complicated lib.

Reply:
make this a trait of the structure.

Reply:
and how do you suggest to do that?